emitter.go

```diff
@@ -61,23 +67,33 @@ func Start(plugins *InOutPlugins, stop chan int) {
 	}
 }
 
+// Close closes all the goroutine and waits for it to finish.
+func Close(quit chan int) {
+	closeOnce.Do(func() {
+		close(quit)
```

Comment:
I think we can change `close(quit)` to `quit <- 1` and remove `closeOnce.Do...`. Can you try it?

Reply:
I think close<-1 defeats the purpose of this PR because we are not waiting for other go routines to stop and the race condition still remains.

Reply:
Makes sense